realm/realm-library/src/main/java/io/realm/MapChangeSet.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import io.realm.internal.OsMapChangeSet;
+
+/**
+ * This interface describes the changes made to a map during the last update.
+ * <p>
+ * {@link MapChangeSet} is passed to the {@link MapChangeListener} which is registered
+ * by {@link RealmMap#addChangeListener(MapChangeListener)} )}.
+ */
+public interface MapChangeSet<T> {
+
+    /**
+     * The number of entries that have been deleted in the previous version of the map.
+     *
+     * @return the number of deletions.
+     */
+    long getDeletionsCount();
+
+    /**
+     * Array containing the keys that have been inserted in the previous version of the map.
+     *
+     * @return array with the keys that have been inserted.
+     */
+    T[] getInsertions();
+
+    /**
+     * Array containing the keys that have been modified in the previous version of the map.
+     *
+     * @return array with the keys that have been modified.
+     */
+    T[] getModifications();
```

Comment:
We might like to have the same function name as for Lists or Objects: `getChanges`? What do you think?

Reply:
Fair enough